test/reply-code.test.js

```diff
@@ -57,3 +57,24 @@ test('code should handle null/undefined/float', t => {
     t.equal(res.statusCode, 404)
   })
 })
+
+test('code should handle 204', t => {
```

Comment:
Can you also add a test where the body is explicitly ignored? e.g. sending a payload but being ignored because `204`

Reply:
but the payload being ignored functionality needs to get implemented right? cause right now if you send `payload` in `204` it throws error, so we need to ignore most of the steps if code is `204` ?

Reply:
Yeah, kind of, you are already partially covered. It should be a matter of calling `onSendHook(this, undefined)` to drop the payload.

This might extend the PR as we will need to document this behaviour.